File: Safari-611.1.10/LayoutTests/http/tests/misc/resources/charset-sniffer-end-sniffing.php
<?php
header("Expires: Thu, 01 Dec 2003 16:00:00 GMT");
header("Cache-Control: no-cache, no-store, must-revalidate");
header("Pragma: no-cache");
header("Content-Type: text/html");

echo "<html><script>parent.firstScript();</script>\n";

// Dump a lot of text to exceed 1024 chars after finishing the head section.
echo "<body>\n";
for ($i = 0; $i < 10000; ++$i)
    echo "a";

// Send off the first portion so it executes before the rest is received.
ob_flush();
flush();
sleep(1);

echo "<html><script>parent.secondScript();</script>\n";
echo "</body></html>\n";
?>

File: LayoutTests/http/tests/misc/resources/charset-sniffer-end-sniffing.py
#!/usr/bin/env python3

import sys
import time

sys.stdout.write(
    'Expires: Thu, 01 Dec 2003 16:00:00 GMT\r\n'
    'Cache-Control: no-cache, no-store, must-revalidate\r\n'
    'Pragma: no-cache\r\n'
    'Content-Type: text/html\r\n\r\n'
    '<html><script>parent.firstScript();</script>\n'
    '<body>\n'
)

for _ in range(0, 10000):
    sys.stdout.write('a')

sys.stdout.flush()
time.sleep(1)

sys.stdout.write(
    '<html><script>parent.secondScript();</script>\n'
    '</body></html>\n'
)